scrapy/resolver.py

```diff
@@ -4,17 +4,25 @@
 from scrapy.utils.datatypes import LocalCache
 
 # TODO: cache misses
-# TODO: make cache size a setting
 
 dnscache = LocalCache(10000)
 
 class CachingThreadedResolver(ThreadedResolver):
+    def __init__(self, reactor, settings):
```

Comment:
I think it would be cleaner to pass cache_enabled(*), cache_size and timeout arguments instead of passing settings object. This way scrapy.resolver won't depend on Scrapy Settings API.

(*) is cache_enabled argument needed here? cache_size=0 could be enough (with all logic kept).